app/src/main/java/org/commcare/commcaresupportlibrary/identity/identity_integration.md

```diff
@@ -0,0 +1,54 @@
+## Integrating CommCare with third party Identity Systems
+
+
+CommCare supports [Android Callouts](https://github.com/dimagi/commcare-android/wiki/Intent-Callout-to-External-Application) based integration with third party Identity Providers that facilitate services like Biometric Enrollment and Identification. Using Android Intent Callouts, it's possible for someone to configure a CommCare app that can call a third party Android based Identity Provider and receive the enrolled case data back into CommCare app. 
+
+If you are an Identity Provider and wants to integrate with CommCare, you can follow this guide to know about Android Intent contracts CommCare supports out of the box. 
+To get access to generic CommCare data models, you need to first add the CommCare support library as a dependency [as described over here](https://github.com/dimagi/commcare-support-library#installation)
+
+
+We define some of the common Identity Provider workflows below and how you can make changes in the Identity Application to support these workflows with CommCare. 
+
+
+#### Enrolling a Beneficiary
+
+You can have CommCare call your app in a form to register biometrics for a beneficiary. This beneficiary enrollment should create a new guid for the beneficiary in the Identity Provider.
+Often you will need this generated guid to be passed back to CommCare so that it can be saved as a case property. In order to do so, you can utilize the `Registration` construct as follows. 
+You can have CommCare call your app in a form to register biometrics for a beneficiary. This beneficiary enrollment should create a new guid for the beneficiary in the Identity Provider. This guid needs to be passed back to CommCare to be saved as a case property as follows 
+
+````
+Intent intent = new Intent();

```

Comment:
I found it very helpful to make these a factory pattern just to make the inline code very self-documenting.

IE:
```
var response = IntentResponseFactory.RegistrationResponse(mandatoryObject).build()

response.finalize(this::Activity)
```

so that the android code is pretty much entirely abstracted away and it's harder for people to make a mistake.